Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -1213,14 +1222,21 @@ void FlexFormattingContext::distribute_any_remaining_free_space()
                 + space_between_items;
 
             if (is_direction_reverse()) {
-                item.main_offset = main_offset - item.main_size - item.margins.main_after - item.borders.main_after - item.padding.main_after;
-                main_offset -= amount_of_main_size_used;
+                item.main_offset = cursor_offset - item.main_size - item.margins.main_after - item.borders.main_after - item.padding.main_after;
+                cursor_offset -= amount_of_main_size_used;
+            } else if (direction == Direction::RightToLeft) {
+                cursor_offset -= amount_of_main_size_used;
+                item.main_offset = cursor_offset + item.margins.main_before + item.borders.main_before + item.padding.main_before;
             } else {
-                item.main_offset = main_offset + item.margins.main_before + item.borders.main_before + item.padding.main_before;
-                main_offset += amount_of_main_size_used;
+                item.main_offset = cursor_offset + item.margins.main_before + item.borders.main_before + item.padding.main_before;
+                cursor_offset += amount_of_main_size_used;
             }
         };
 
+        if (direction == Direction::RightToLeft) {
+            flex_line.items.reverse();
+        }
+
         if (is_direction_reverse()) {
```

Comment:
How does your `Direction` relate to `is_direction_reverse()`? It's a little confusing since there looks like there's some double reserve here. Also you can probably find a way to use the `in_reverse()` iterator, rather than actually reverse the list.

Reply:
so I believe there's some kind of flex directive for reverse as well as the general notion of flex-end that I am trying to implement here. Essentially what this implementation does is move the `offset` to the end and render the elements backwards shifting the cursor left.